src/dodal/beamlines/i22.py

```diff
@@ -23,27 +29,81 @@
 from dodal.log import set_beamline as set_log_beamline
 from dodal.utils import BeamlinePrefix, get_beamline_name, skip_device
 
-BL = get_beamline_name("i22")
+LAB_NAME = BEAMLINE_LAB_MAPPING.get("i22", "i22")
+
+BL = LAB_NAME if LAB_FLAG else get_beamline_name("i22")
+print("BL NAME: ", BL)
 set_log_beamline(BL)
 set_utils_beamline(BL)
 
+IS_LAB = LAB_FLAG and BL == LAB_NAME
+print("is this lab? : ", IS_LAB)
+
 # Currently we must hard-code the visit, determining the visit at runtime requires
 # infrastructure that is still WIP.
 # Communication with GDA is also WIP so for now we determine an arbitrary scan number
 # locally and write the commissioning directory. The scan number is not guaranteed to
 # be unique and the data is at risk - this configuration is for testing only.
-set_directory_provider(
+directory_provider = (
     StaticVisitDirectoryProvider(
+        BL,
+        Path("/dls/p38/data/2024/cm37282-2/bluesky"),
+        client=LocalDirectoryServiceClient(),
+    )
+    if IS_LAB
+    else StaticVisitDirectoryProvider(
         BL,
         Path("/dls/i22/data/2024/cm37271-2/bluesky"),
         client=DirectoryServiceClient("http://i22-control:8088/api"),
     )
 )
 
 
+# d3 at p38
+def oav(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> AravisDetector:
+    return device_instantiation(
+        AravisDetector if IS_LAB else NXSasOAV,
+        "d3" if IS_LAB else "oav",
+        f"-DI-{'DCAM' if IS_LAB else 'OAV'}-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        drv_suffix="DET:",
+        hdf_suffix="HDF5:",
+        directory_provider=get_directory_provider(),
+    )
+
+
+set_directory_provider(directory_provider)
+
+
+# d11 at p38, but disconnected
+@skip_device(lambda: BL == LAB_NAME)
 def saxs(
     wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> PilatusDetector:
+) -> AravisDetector | NXSasPilatus:
+    if IS_LAB:
+        return device_instantiation(
+            AravisDetector,
+            "d11",
+            "-DI-DCAM-03:",
+            wait_for_connection,
+            fake_with_ophyd_sim,
+            drv_suffix="DET:",
+            hdf_suffix="HDF5:",
+            # metadata_holder=NXSasMetadataHolder(
+            #     x_pixel_size=(1.72e-1, "mm"),
+            #     y_pixel_size=(1.72e-1, "mm"),
+            #     description="Dectris Pilatus3 2M",
+            #     type="Photon Counting Hybrid Pixel",
+            #     sensor_material="silicon",
+            #     sensor_thickness=(0.45, "mm"),
+            #     distance=(4711.833684146172, "mm"),
+            # ),
```

Comment:
Define a `NXSasAravis` that takes the metadataholder  so processing can reliably treat `saxs` as a valid NXsas detector, and find the values.

Reply:
done